source/patterns/@aws-solutions-constructs/aws-lambda-eventbridge/lib/index.ts

```diff
@@ -0,0 +1,135 @@
+  /**
+   * Existing instance of a custom EventBus.
+   *
+   * @default - None
+   */
+  readonly existingEventBusInterface?: events.IEventBus;
+  /**
+   * A new custom EventBus is created with provided props.
+   *
+   * @default - None
+   */
+  readonly eventBusProps?: events.EventBusProps;
+  /**
+   * An existing VPC for the construct to use (construct will NOT create a new VPC in this case)
+   */
+  readonly existingVpc?: ec2.IVpc;
+  /**
+   * Properties to override default properties if deployVpc is true
+   */
+  readonly vpcProps?: ec2.VpcProps;
+  /**
+   * Whether to deploy a new VPC
+   *
+   * @default - false
+   */
+  readonly deployVpc?: boolean;
+  /**
+   * Optional Name for the EventBus's name environment variable set for the Lambda function.
+   *
+   * @default - EVENTBUS_NAME is used
+   */
+  readonly eventBusEnvironmentVariableName?: string;
+}
+
+/**
+ * @summary The LambdaToEventbridge class.
+ */
+export class LambdaToEventbridge extends Construct {
+  public readonly lambdaFunction: lambda.Function;
+  public readonly eventBus?: events.IEventBus;
+  public readonly vpc?: ec2.IVpc;
+
+  /**
+   * @summary Constructs a new instance of the LambdaToEventbridge class.
+   * @param {cdk.App} scope - represents the scope for all the resources.
+   * @param {string} id - this is a a scope-unique id.
+   * @param {LambdaToEventbridgeProps} props - user provided props for the construct.
+   * @since 1.120.0
+   * @access public
+   */
+  constructor(scope: Construct, id: string, props: LambdaToEventbridgeProps) {
+    super(scope, id);
+    defaults.CheckProps(props);
+
+    if (props.deployVpc || props.existingVpc) {
+      if (props.deployVpc && props.existingVpc) {
+        throw new Error("More than 1 VPC specified in the properties");
+      }
+
+      this.vpc = defaults.buildVpc(scope, {
+        defaultVpcProps: defaults.DefaultIsolatedVpcProps(),
+        existingVpc: props.existingVpc,
+        userVpcProps: props.vpcProps,
+        constructVpcProps: {
+          enableDnsHostnames: true,
+          enableDnsSupport: true,
+        },
+      });
+      defaults.AddAwsServiceEndpoint(scope, this.vpc, defaults.ServiceEndpointTypes.EVENTS);
+    }
+
+    // Setup the Lambda function
+    this.lambdaFunction = defaults.buildLambdaFunction(this, {
+      existingLambdaObj: props.existingLambdaObj,
+      lambdaFunctionProps: props.lambdaFunctionProps,
+      vpc: this.vpc,
+    });
+
+    // Build Event Bus
+    this.eventBus = defaults.buildEventBus(this, {
+      existingEventBusInterface: props.existingEventBusInterface,
+      eventBusProps: props.eventBusProps
+    });
+
+    // Configure environment variables
+    const eventBusEnvironmentVariableName = props.eventBusEnvironmentVariableName || 'EVENTBUS_NAME';
+    const eventBusName = this.eventBus?.eventBusName || 'default';
+    this.lambdaFunction.addEnvironment(eventBusEnvironmentVariableName, eventBusName);
```

Comment:
In the lambda function, when sending an event to an event bus do you actually use the name "default" when using the default bus or do you not send any name?

If the latter, seems like a better solution is either setting this environment variable to a null value or not setting it at all. 

OK, a quick test reveals that every environment variable must have a value. Let's consider not setting the variable if a lambda function doesn't send a name when targeting the default bus.

Reply:
sdk expects an eventbus name in case lambda is trying to publish to a custom eventbus. For publishing to default eventBus, name is optional, but it doesn't hurt sending passing it.